apps/gitness/src/components/Header.tsx

```diff
@@ -1,47 +1,193 @@
+
+const CREATE_PROJECT = -2
+const CreateProjectDropdownOptions = { value: CREATE_PROJECT, label: 'Create project' }
 
 export default function Header() {
   const navigate = useNavigate()
   const space = useGetSpaceURLParam()
+  const repoRef = useGetRepoRef()
+  const { pipelineId, repoId, executionId: executionIdParam } = useParams()
+  const executionId = executionIdParam ? parseInt(executionIdParam) : undefined
+
   const { spaces } = useAppContext()
-  const [projects, setProjects] = useState<Project[]>([{ id: 'create-project', name: 'Create project' }])
-  const [selectedProject, setSelectedProject] = useState<Project | undefined>(undefined)
+  const [selectedProject, setSelectedProject] = useState<ProjectDropdownOption | undefined>()
+
+  const [projects, setProjects] = useState<ProjectDropdownOption[]>([CreateProjectDropdownOptions])
+  const [repositoryOptions, setRepositoryOption] = useState<RepositoryDropdownOption[] | undefined>()
+  const [pipelineOptions, setPipelineOptions] = useState<PipelineDropdownOption[] | undefined>()
+  const [executionOptions, setExecutionOptions] = useState<ExecutionDropdownOption[] | undefined>()
 
   useEffect(() => {
     if (spaces.length > 0) {
-      setProjects((existingProjects: Project[]) =>
-        /* dedupe items */
-        Array.from(
-          new Map(
-            [...spaces.map(space => ({ id: space.id, name: space.path })), ...existingProjects].map(project => [
-              project.id,
-              project
-            ])
-          ).values()
-        )
-      )
+      setProjects([
+        ...(spaces?.map(space => ({ label: space.path as string, value: space.id as number })) ?? []),
+        ...[CreateProjectDropdownOptions]
+      ])
+    } else {
+      setProjects([CreateProjectDropdownOptions])
     }
   }, [spaces])
 
-  useEffect(() => setSelectedProject(projects.find(item => item.name === space)), [space, projects])
+  useEffect(() => setSelectedProject(projects.find(item => item.label === space)), [space, projects])
 
-  return (
-    <TopBarWidget
-      projects={projects}
-      onSelectProject={(project: Project) => {
+  const { refetch: fetchRepos } = useListReposQuery({ queryParams: {}, space_ref: `${space}/+` }, { enabled: false })
+
+  // load repositories
+  useEffect(() => {
+    if (space && repoId) {
+      fetchRepos().then(response => {
+        setRepositoryOption(
+          response.data?.body?.map(repo => ({ label: repo.identifier ?? '', value: repo.identifier ?? '' }))
+        )
+      })
+    } else {
+      setRepositoryOption(undefined)
+    }
+  }, [space, repoId])
+
+  const { refetch: fetchPipelines } = useListPipelinesQuery(
+    {
+      repo_ref: repoRef,
+      queryParams: {}
+    },
+    {
+      enabled: false
+    }
+  )
+
+  // load pipelines
+  useEffect(() => {
+    if (repoRef) {
+      fetchPipelines().then(response => {
+        setPipelineOptions(
+          response.data?.body?.map(pipeline => ({
```

Comment:
this mapping can be done when rendering the component, instead of putting it in state